services/QuillLMS/client/app/bundles/Diagnostic/components/questions/response.tsx

```diff
@@ -0,0 +1,446 @@
+import * as React from 'react'
+import * as jsDiff from 'diff'
+
+import { ContentState, EditorState } from 'draft-js'
+import _ from 'underscore'
+
+import {
+  Modal,
+  TextEditor
+} from '../../../Shared/index'
+import massEdit from '../../actions/massEdit'
+import questionActions from '../../actions/questions'
+import {
+  deleteResponse,
+  getGradedResponsesWithCallback,
+  submitResponseEdit,
+} from '../../actions/responses'
+import sentenceFragmentActions from '../../actions/sentenceFragments'
+import C from '../../constants'
+import { rematchOne } from '../../libs/grading/rematching'
+import { getStatusForResponse } from '../../../Shared/index'
+
+import ConceptSelectorWithCheckbox from '../shared/conceptSelectorWithCheckbox.jsx'
+
+import ResponseList from './responseList'
+import { Question, Response } from '../../libs/grading/rematching'
+
+interface ResponseProps {
+  allExpanded: boolean,
+  ascending: boolean,
+  concepts: Array<Object>,
+  dispatch: Function,
+  expand: Function,
+  expanded: boolean,
+  getChildResponses: Function,
+  getResponse: Function,
+  mode: string,
+  massEditResponses: Object,
+  passedResponse: Response,
+  question: Question,
+  questionID: string,
+  readOnly: boolean,
+  responses: Array<Response>,
+  state: Object,
+  states: Object,
+  updateParentResponse: Function,
+}
+
+
+const Response = ({allExpanded, ascending, concepts, dispatch, expand, expanded, getChildResponses, getResponse, massEditResponses, mode, passedResponse, question, questionID, readOnly, responses, state, states, updateParentResponse}: ResponseProps) => {
+  const [response, setResponse] = React.useState(passedResponse)
+  const [feedback, setFeedback] = React.useState(passedResponse.feedback || '')
+  const [selectedBoilerplate, setSelectedBoilerplate] = React.useState('')
+  const [parent, setParent] = React.useState(null)
+
+  let conceptResultsTemp = {}
+  if (passedResponse.concept_results) {
+    if (typeof passedResponse.concept_results === 'string') {
+      conceptResultsTemp = JSON.parse(passedResponse.concept_results)
+    } else {
+      conceptResultsTemp = passedResponse.concept_results
+    }
+  }
+  const [conceptResults, setConceptResults] = React.useState(conceptResultsTemp)
+  const [actions, setActions] = React.useState(mode === 'sentenceFragment' ? sentenceFragmentActions : questionActions)
+  const [statusCode, setStatusCode] = React.useState(getStatusForResponse(passedResponse))
+  const newResponseOptimal = React.useRef(null)
+
+  React.useEffect(() => {
+    setFeedback(response.feedback)
+    setStatusCode(getStatusForResponse(response))
+
+    const { concept_results, } = response;
+    let conceptResults = {}
+    if (concept_results) {
+      if (typeof concept_results === 'string') {
+        conceptResults = JSON.parse(concept_results)
+      } else {
+        conceptResults = concept_results
+      }
+    }
+    setConceptResults(conceptResults)
```

Comment:
Once you have the function above, you can replace this code with an invocation of that as well.